packages/gas-fee-controller/src/GasFeeController.ts

```diff
@@ -443,10 +447,23 @@ export class GasFeeController extends StaticIntervalPollingController<GasFeePoll
       decimalChainId: number;
 
     if (networkClientId !== undefined) {
-      const networkClient = this.messagingSystem.call(
-        'NetworkController:getNetworkClientById',
-        networkClientId,
-      );
+      let networkClient;
+      try {
+        networkClient = this.messagingSystem.call(
+          'NetworkController:getNetworkClientById',
+          networkClientId,
+        );
+      } catch (error) {
+        if (error instanceof NoNetworkClientFoundError) {
+          log(error.message);
```

Comment:
I figured that we ought to be aware of the problem on some level, I am just unsure what the best place is.

@Gudahtt What would be your recommendation here? I seem to recall a comment you made in some PR a while back about switching away from using `console.error` / `console.warn`. There's also [this PR](https://github.com/MetaMask/core/pull/4440) where we introduced a separate production-level logger, but it seems like this log message would continue to surface in Sentry which we might not want.

Reply:
The pattern in the linked PR seems ideal to me because the client can choose whether it's enabled or not. That was generally the goal: let the client control logging, don't make it mandatory.

Logs can definitely be useful as Sentry breadcrumbs.

Reply:
About this specific scenario, can we make this situation impossible? I don't quite follow how it could happen.

Is there an active poll on a deleted network that we haven't shut down, or is something else trying to get estimates from a non-existent network? I'd think the root problem lies elsewhere.

Reply:
Yeah, I guess you're right. Maybe hiding the error isn't such a good thing in this case.